src/Components/Aspire.Azure.AI.Inference/AspireAzureAIInferenceExtensions.cs

```diff
@@ -0,0 +1,205 @@
+                    {
+                        var credential = new AzureKeyCredential(settings.Key);
+                        return new ChatCompletionsClient(settings.Endpoint, credential, options);
+                    }
+                    else
+                    {
+                        return new ChatCompletionsClient(settings.Endpoint, settings.TokenCredential ?? new DefaultAzureCredential(), options);
+                    }
+                }
+            });
+        }
+
+        protected override void BindClientOptionsToConfiguration(IAzureClientBuilder<ChatCompletionsClient, AzureAIInferenceClientOptions> clientBuilder, IConfiguration configuration)
+        {
+#pragma warning disable IDE0200 // Remove unnecessary lambda expression - needed so the ConfigBinder Source Generator works
+            clientBuilder.ConfigureOptions(options => configuration.Bind(options));
+#pragma warning restore IDE0200
+        }
+
+        protected override void BindSettingsToConfiguration(ChatCompletionsClientSettings settings, IConfiguration configuration)
+            => configuration.Bind(settings);
+
+        protected override IHealthCheck CreateHealthCheck(ChatCompletionsClient client, ChatCompletionsClientSettings settings)
+            => throw new NotImplementedException();
+
+        protected override bool GetHealthCheckEnabled(ChatCompletionsClientSettings settings)
+            => false;
+
+        protected override bool GetMetricsEnabled(ChatCompletionsClientSettings settings)
+            => !settings.DisableMetrics;
+
+        protected override TokenCredential? GetTokenCredential(ChatCompletionsClientSettings settings)
+            => settings.TokenCredential;
+
+        protected override bool GetTracingEnabled(ChatCompletionsClientSettings settings)
+            => !settings.DisableTracing;
+    }
+
+    /// <summary>
+    /// Creates a <see cref="IChatClient"/> from the <see cref="ChatCompletionsClient"/> registered in the service collection.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireChatCompletionsClientBuilder" />.</param>
+    /// <returns></returns>
+    public static ChatClientBuilder AddChatClient(this AspireChatCompletionsClientBuilder builder)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        return builder.HostBuilder.Services.AddChatClient(
+            services => CreateInnerChatClient(builder, services));
+    }
+
+    /// <summary>
+    /// Creates a <see cref="IChatClient"/> from the <see cref="ChatCompletionsClient"/> registered in the service collection.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireChatCompletionsClientBuilder" />.</param>
+    /// <param name="serviceKey">The service key with which the <see cref="IChatClient"/> will be registered.</param>
+    /// <returns></returns>
+    public static ChatClientBuilder AddKeyedChatClient(this AspireChatCompletionsClientBuilder builder, string? serviceKey)
```

Comment:
`serviceKey` shouldn't be nullable if you are calling `AddKeyedChatClient`.

Reply:
The builder has a service key:

```cs
serviceKey = serviceKey ?? builder.ServiceKey;
```